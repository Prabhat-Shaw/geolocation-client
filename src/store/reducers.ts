/**
 * Combine all reducers in this file and export the combined reducers.
 */

import { combineReducers, Reducer } from '@reduxjs/toolkit';
import { authenticationReducer } from 'app/components/Authentication/slice';
import { AuthenticationState } from 'app/components/Authentication/slice/types';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { InjectedReducersType } from 'utils/types/injector-typings';

const authenticationPersistConfig = {
  key: 'auth',
  storage,
  whitelist: ['isAuthenticated', 'user'],
};

/**
 * Merges the main reducer with the router state and dynamically injected reducers
 */
export function createReducer(injectedReducers: InjectedReducersType = {}) {
  return combineReducers({
    ...injectedReducers,
    authentication: persistReducer(
      authenticationPersistConfig,
      authenticationReducer as Reducer<AuthenticationState>,
    ),
  });
}
