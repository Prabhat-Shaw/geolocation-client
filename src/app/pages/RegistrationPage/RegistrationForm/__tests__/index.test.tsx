import { render } from '@testing-library/react';
import * as React from 'react';
import { Provider } from 'react-redux';
import { configureAppStore } from 'store/configureStore';
import { RegistrationForm } from '..';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

describe('<RegistrationForm  />', () => {
  it('should match snapshot', () => {
    const store = configureAppStore();

    const loadingIndicator = render(
      <Provider store={store}>
        <RegistrationForm />
      </Provider>,
    );
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});
