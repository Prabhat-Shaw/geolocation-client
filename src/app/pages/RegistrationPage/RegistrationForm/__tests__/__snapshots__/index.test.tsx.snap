// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationForm  /> should match snapshot 1`] = `
<div
  class="RegistrationForm__Div-sc-riav6x-0"
>
  
  <form>
    <input
      name="emailAddress"
    />
    <input
      name="password"
    />
    <input
      type="submit"
    />
  </form>
</div>
`;
